torch_geometric/nn/models/graph_unet.py

```diff
@@ -0,0 +1,112 @@
+import torch
+import torch.nn.functional as F
+from torch_sparse import spspmm
+from torch_geometric.nn import TopKPooling, GCNConv
+from torch_geometric.utils import (add_self_loops, sort_edge_index,
+                                   remove_self_loops)
+from torch_geometric.utils.repeat import repeat
+
+
+class GraphUNet(torch.nn.Module):
+    r"""The Graph U-Net model from the `"Graph U-Nets"
+    <https://arxiv.org/abs/1905.05178>`_ paper which implements a U-Net graph
+    architecture with graph pooling and unpooling operations.
+
+    Args:
+        channels (int): Size of each sample.
+        depth (int): The depth of the U-Net architecture.
+        pool_ratios (float or [float], optional): Graph pooling ratio for each
+            depth. (default: :obj:`0.5`)
+        sum_res (bool, optional): If set to :obj:`False`, will use
+            concatenation for integration of skip connections instead
+            summation. (default: :obj:`True`)
+        act (torch.nn.functional, optional): The nonlinearity to use.
+            (default: :obj:`torch.nn.functional.relu`)
+    """
+
+    def __init__(self, channels, depth, pool_ratios=0.5, sum_res=True,
+                 act=F.relu):
+        super(GraphUNet, self).__init__()
+
+        self.channels = channels
+        self.depth = depth
+        self.pool_ratios = repeat(pool_ratios, depth)
+
+        self.act = act
+        self.sum_res = sum_res
+
+        self.pools = torch.nn.ModuleList()
+        self.down_convs = torch.nn.ModuleList()
+        for i in range(depth):
+            self.pools.append(TopKPooling(channels, self.pool_ratios[i]))
```

Comment:
Number of pooling should be depth-1, I think the issue is that you are ppolign before the first convolution.